subprojects/core/src/main/java/org/gradle/api/internal/tasks/properties/annotations/NestedBeanPropertyAnnotationHandler.java

```diff
@@ -38,36 +31,5 @@
 
     @Override
     public void visitPropertyValue(final PropertyValue propertyValue, PropertyVisitor visitor, PropertySpecFactory specFactory) {
-        DefaultTaskInputPropertySpec propertySpec = specFactory.createInputPropertySpec(propertyValue.getPropertyName() + ".class", new NestedPropertyValue(propertyValue));
-        propertySpec.optional(propertyValue.isOptional());
-        visitor.visitInputProperty(propertySpec);
-    }
-
-    private static class NestedPropertyValue implements ValidatingValue {
-        private final PropertyValue propertyValue;
-
-        public NestedPropertyValue(PropertyValue propertyValue) {
-            this.propertyValue = propertyValue;
-        }
-
-        @Nullable
-        @Override
-        public Object call() {
-            Object value = propertyValue.getValue();
-            return value == null ? null : value.getClass().getName();
-        }
-
-        @Override
-        public void validate(String propertyName, boolean optional, ValidationAction valueValidator, TaskValidationContext context) {
-            Object value = propertyValue.getValue();
-            if (value == null) {
-                if (!optional) {
-                    String realPropertyName = propertyName.substring(0, propertyName.length() - ".class".length());
-                    context.recordValidationMessage(ERROR, String.format("No value has been specified for property '%s'.", realPropertyName));
-                }
-            } else {
-                valueValidator.validate(propertyName, value, context, ERROR);
-            }
-        }
```

Comment:
This is weird. We don't need this handler anymore? If so, we can use `NoOpPropertyAnnotationHandler` to handle `@Nested`. But it seems weird that we don't need this guy.

Reply:
The only thing this annotation handler did was to add `.class` input properties. This is now handled completely by `DefaultPropertyWalker`.

We either have the nested bean handling done by property walker or somehow allow the `NestedBeanPropertyAnnotationHandler` to add some nodes to the queue for walking. I think having the logic in `DefaultPropertyWalker` makes sense to me.